powerscale/provider/writable_snapshots_resource_test.go

```diff
@@ -0,0 +1,267 @@
+/*
```

Comment:
The following is the complete code with improvements:

```python
'''
Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.

Licensed under the Mozilla Public License Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://mozilla.org/MPL/2.0/

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
'''

package provider

import (
	"context"
	"fmt"
	"regexp"
	"terraform-provider-powerscale/client"
	"terraform-provider-powerscale/powerscale/helper"
	"testing"
)

import (
	"github.com/bytedance/mockey"
	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
	"github.com/hashicorp/terraform-plugin-testing/terraform"
)

var writableSnapshotResourceName = "powerscale_writable_snapshot.test"

func TestAccwritableSnapshotResourceImport(t *testing.T) {
	resource.Test(t, resource.TestCase{
		PreCheck:                 func() { testAccPreCheck(t) },
		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
		Steps: []resource.TestStep{
			{
				Config: ""
				SimpleResourceConfig + writableSnapshotResourceConfig,
				Check: resource.ComposeAggregateTestCheckFunc(
					resource.TestCheckResourceAttr(writableSnapshotResourceName, "dst_path", "/ifs/abcd"),
				),
			},
			{
				PreConfig: func() {
					FunctionMocker = mockey.Mock(helper.GetWritableSnapshot).Return(nil, fmt.Errorf("mock error")).Build()
				},
				Config:       writableSnapshotResourceConfig,
				ResourceName: writableSnapshotResourceName,
				ImportState:  true,
				ExpectError:  regexp.MustCompile(`.*mock error*.`),
			},
			// Import testing
			{
				PreConfig: func() {
					FunctionMocker.Release()
					FunctionMocker = mockey.Mock(helper.GetWritableSnapshot).Return(nil, nil).Build()
				},
				Config:       writableSnapshotResourceConfig,
				ResourceName: writableSnapshotResourceName,
				ImportState:  true,
				ExpectError:  nil,
				ImportStateCheck: func(s []*terraform.InstanceState) error {
					resource.TestCheckResourceAttrSet(writableSnapshotResourceName, "dst_path")
					resource.TestCheckResourceAttrSet(writableSnapshotResourceName, "src_path")
					resource.TestCheckResourceAttrSet(writableSnapshotResourceName, "snap_id")
					return nil
				},
			},
		},
	})
}
```

This answer directly follows the provided input in the GitHub diff format, suggesting improvements based on the given diff without requiring the full code.